.cbmc-batch/jobs/aws_cryptosdk_compare_hash_elems_by_key_string/aws_cryptosdk_compare_hash_elems_by_key_string_harness.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/private/utils.h>
+#include <make_common_data_structures.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/utils.h>
+
+void aws_cryptosdk_compare_hash_elems_by_key_string_harness() {
+    /* Non-deterministic inputs. */
+    struct aws_hash_element *elem_a = can_fail_malloc(sizeof(*elem_a));
+    if (elem_a != NULL) {
+        if (nondet_bool()) {
+            elem_a->key = ensure_string_is_allocated_bounded_length(MAX_STRING_LEN);
+            __CPROVER_assume(aws_string_is_valid(elem_a->key));
+        } else {
+            elem_a->key = NULL;
+        }
+    }
```

Comment:
This code is duplicated - might be worth factoring out into a function

Reply:
I know it is repetitive here, but this code is specific as well. It creates a hash element that stores a bounded string object (we could create the same thing for different objects). To avoid duplicity, I wrote a function, but I added it to the harness itself. We could add to AWS C Common a function for that (since it's related to the `aws_hash` module). What do you think?